main.py

```diff
@@ -2,23 +2,38 @@
 #
 # Simple manager prototype for xqemu
 #
-from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox
+from PyQt5.QtWidgets import QApplication, QDialog, QFileDialog, QMainWindow, QMessageBox, QTableWidgetItem, QAbstractItemView, QHeaderView
 from PyQt5.uic import loadUiType
 from PyQt5 import QtCore, QtGui
 from qmp import QEMUMonitorProtocol
+from glob import glob1
 import sys
 import os, os.path
 import json
 import subprocess
 import time
 import platform
+import sys
 
 SETTINGS_FILE = './settings.json'
 
 # Load UI files
 settings_class, _ = loadUiType('settings.ui')
 mainwindow_class, _ = loadUiType('mainwindow.ui')
 
+def read_from_hex_offset(file, hex_offset):
+	file.seek(hex_offset)
+	return file.read(32)
+
+def getFormat(fileformats, settings, filename):
+	for element in fileformats:
+		for offset in element['offset']:
+			file = open(settings.settings['iso_path'] + '/' + filename, 'rb')
```

Comment:
I suggest to give this function an absolute path.

So if we allow an "open-from-file" from outside of the ISO path, we can still use it.